static/js/hashchange.js

```diff
@@ -118,8 +118,33 @@ function is_overlay_hash(hash) {
     return overlay_list.includes(main_hash);
 }
 
+function attempt_render_default_view() {
+    // Returns true if successful
+    // We only allow all_messages and recent_topics
+    // to be rendered without a hash.
+    // NOTE: Setting a hash which is not rendered on
+    // empty hash (like a stream narrow) will
+    // introduce a bug that user will not be able to
+    // go back in broswer history. See
+    // https://chat.zulip.org/#narrow/stream/9-issues/topic/Browser.20back.20button.20on.20RT
+    // for detailed description of the issue.
+    if (page_params.default_view === "recent_topics") {
+        recent_topics.show();
+        return true;
+    }
+
+    if (page_params.default_view === "all_messages") {
+        activate_home_tab();
+        return true;
+    }
+
+    return false;
+}
+
 export function show_default_view() {
-    window.location.hash = page_params.default_view;
+    if (!attempt_render_default_view()) {
+        window.location.hash = page_params.default_view;
+    }
 }
```

Comment:
I think one of the callers (but not the other) expects us to change the hash from `#narrow/` to `#`?
```
                // If the narrow URL didn't parse, clear                                             
                // window.location.hash and send them to the home tab                                
                show_default_view();                                                                 
```